lms/static/scripts/frontend_apps/components/StudentSelector.tsx

```diff
@@ -7,40 +7,51 @@ import {
 } from '@hypothesis/frontend-shared/lib/next';
 import classnames from 'classnames';
 
-export type Student = {
-  displayName: string;
-};
+import type { StudentInfo } from '../config';
+
+export type StudentOption = Pick<StudentInfo, 'displayName' | 'lmsId'>;
 
 export type StudentSelectorProps = {
   /** Callback invoked when the selected student changes */
-  onSelectStudent: (index: number) => void;
-  /** Index of selected student in `students` or -1 if no student is selected */
-  selectedStudentIndex: number;
-  /** Ordered list of students to display in the drop-down */
-  students: Student[];
+  onSelectStudent: (lmsId: string | null) => void;
+  selectedStudent: StudentOption | null;
+  students: StudentOption[];
 };
 
 /**
  * A drop-down control that allows selecting a student from a list of students.
  */
 export default function StudentSelector({
   onSelectStudent,
-  selectedStudentIndex,
+  selectedStudent,
   students,
 }: StudentSelectorProps) {
-  // Disable the next button if at the end of the list. The length is equal to
-  // the student list plus the default "All Students" option.
-  const hasNextView = selectedStudentIndex + 1 < students.length;
-  // Disable the previous button only if the selectedStudentIndex is less than 0
-  // indicating the "All Students" choice is selected.
-  const hasPrevView = selectedStudentIndex >= 0;
+  const options: StudentOption[] = [
+    { lmsId: 'none', displayName: 'All Students' },
```

Comment:
You could set `lmsId` to `null` here to indicate that the option has no corresponding entry. This would require adjusting the `StudentOption` type.